test/collection_form_test.rb

```diff
@@ -112,14 +112,31 @@ def initialize(house)
       "0" => { "room_name" => "Kitchen", "area" => 18 },
     })
     @house.reload.city.must_equal "Fancyville"
+
+    # Update existing record
     @house.house_rooms.size.must_equal 3
     @rooms.last.reload.name.must_equal "Lounge"
     @rooms.last.area.must_equal 9
+
+    # Create new record
     room = @house.house_rooms.order(:id).last
     room.name.must_equal "Kitchen"
     room.area.must_equal 18
   end
 
+  it "returns false from valid? if a validation fails on the child records" do
```

Comment:
Could we maybe add a test for validations that are defined on the form (as opposed to the backing records) failing?

Reply:
Added new test it block `returns false from valid? if a validation fails on the child form validation`"